# Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

## 1. Известно, что генеральная совокупность распределена нормально
## со средним квадратическим отклонением, равным 16.
## Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

$$T_{1,2} = \overline{X} \pm \frac{s_0}{\sqrt{n}} \cdot c_{\gamma}$$

где $T_1$, $T_2$ — нижняя и верхняя границы доверительного интервала,

$\overline{X}=M=80$  — выборочное среднее арифметическое ,  
$s_0=16$ — среднее квадратичное отклонение по выборке (несмещенное),

$n=256$ — размер выборки,

$\gamma=0.95$ — доверительная вероятность.  

$c_{\gamma}$ найдем из соотношения  $Ф(с_{\gamma})= \frac{\gamma}{2}=0.475$ по Таблице значений функции Лапласа $c_{\gamma}=1.96$

$$T_1 = \overline{X} + \frac{s_0}{\sqrt{n}} \cdot c_{\gamma}=80+\frac{16}{\sqrt{256}}\cdot1.96 = 80+1.96=81.96$$

$$T_2 = \overline{X} - \frac{s_0}{\sqrt{n}} \cdot c_{\gamma}=80-\frac{16}{\sqrt{256}}\cdot1.96 = 80-1.96=78.04$$


## 2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 
## 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
## Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

Нам необходимо оценить вероятность того что истинное значение величины $Х$ генеральной совокупности - математическое ожидание $а$, попадет в дверительный интервал с вероятностью $\gamma=0.95$, если нам известна выборочная средняя величина $\bar{x}$ и выборочная дисперсия $D_x$(тоесть была произведена точечная оценка и о самой генеральной совокупности нам ничего не известно, короме того что она распределена нормально)  
для оценки будем использовать формулу $$\bar{x}-t_{\gamma}\frac{s}{\sqrt{n}}<a<\bar{x}+t_{\gamma}\frac{s}{\sqrt{n}}$$  
Поскольку выборка мала, нам еще необходимо будет для корректной оценки исправленное среднеквадратичное отклонение $s$ - то есть посчитать несмещенную дисперсию по формуле $s=\frac{n}{n-1}\cdot D_x$

Считаем:  

Выборочная средняя величина:  
$\bar{x}=\frac{1}{n}\sum x_i = \frac{1}{10}\cdot(6.9+6.1+6.2+6.8+7.5+6.3+6.4+6.9+6.7+6.1)=6.59$

In [23]:
n=10
xi=(6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1)
x_=(1/n)*sum(xi)
x_

6.59

Выборочная дисперсия:  
$D_x=\frac{1}{n}\sum(x_i-\bar{x})^2 =0.183$

In [26]:
sum_2=0
for i in xi:
    sum_2+=(i-x_)**2
D=(1/10)*sum_2
D

0.1829000000000001

Выборочное среднеквадратичное несмещенное отклонение  
$s=\frac{n}{n-1}\cdot\sqrt{D_x}=\frac{10}{9}\sqrt{0.1829}=0.475$

In [30]:
(10/9)*(0.183**0.5)

0.47531665858238753

$t_{\gamma}$ определяется по таблице Стьюдента $t_{\gamma}(10-1,(1-0.95)/2)=2.262$

Вычислим точность оценки:  
    $\sigma=\frac{t_{\gamma}s}{\sqrt{n}}=\frac{2.262\cdot0.475}{\sqrt{10}}=0.34$

In [32]:
(2.262*0.475)/(10**0.5)

0.3397709231967915

Получаем диапозон:  

$6.59-0.34<a<6.59+0.34$  


$6.25<a<6.93$  

в данный интервал с вероятностью 0.95 попадает математическое ожидание(истинное значение) $a$ измеряемой величины генеральной совокупности $X$

## 3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
## Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

Критерий $T=\frac{(\bar{x_в}-a_0)\sqrt{n-1}}{S}$ - случайная величина, имеющая распределение Стьюдента.  

$H_0: a_0=a=17$  
$H_1: x_в>a_0=17.5$  

по всей видимости дисперсия относиться к случайной выборке, поэтому посчитаем несмещенное среднеквадратичное отклонение $S$  
$S=\frac{100}{100-1}\cdot \sqrt{4}=2.0202$

In [41]:
(100/99)*(4**0.5)

2.0202020202020203

$T_{набл}=\frac{(17.5-17)\sqrt{100-1}}{2.0202}=2.462$

$t_{пркр}=2.01$  

$t_{пркр}<T_{набл}\Rightarrow H_0$ - отвергается

In [43]:
((17.5-17)*(99)**0.5)/2.0202

2.462596369435254

## 4. Продавец утверждает, что средний вес пачки печенья составляет 200 г.Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
## 202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
## Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

Критерий $T=\frac{(\bar{X}-a_0)\sqrt{n-1}}{S}$ - случайная величина, имеющая распределение Стьюдента.  

$H_0: a_0=a=200$  
$H_1: x_в\neq a_0=198.5$  

In [44]:
n=10
xi=(202, 203, 199, 197, 195, 201, 200, 204, 194, 190)
x_=(1/n)*sum(xi)
sum_2=0
for i in xi:
    sum_2+=(i-x_)**2
D=(1/n)*sum_2
s=(n/(n-1))*(D**0.5)
print(f"x_={x_}, D={D}, s={s}")

x_=198.5, D=17.85, s=4.69436226095058


$T_{набл}=\frac{(198.5-200)\sqrt{10-1}}{4.6943622}=-0.958$

In [46]:
((198.5-200)*3)/4.6943622

-0.9585966758167915

In [48]:
(2720-2900)/(700/(49**0.5))

-1.8

$t_{лвкр}[(10-1),0.99/2]=-3.25$  
https://infotables.ru/matematika/108-kvantili-protsentili-raspredeleniya-styudenta-ili-koeffitsienty-styudenta-tablitsa

$t_{лвкр}<T_{набл}\Rightarrow H_0$ - подтверждается - утверждение продавца верное (о том что средняя пачка весит 200 гр)

## 5. На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

$$\frac{190-178}{\sqrt{D(X)}}=\frac{12}{5}=2.4$$